### PR TITLE
fix: Remove "copy" from copied JS object name/action name

### LINKED ISSUE #15143: [Bug]: Remove "copy" from copied JS object name
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Description

A user pointed out that on copying the JS object from the JS editor page, we are appending "copy" to the name, but on copying it from the explorer page, we don't do this.

https://app.intercom.com/a/apps/y10e7138/inbox/inbox/conversation/164629100140873#part_id=comment-164629100140873-15716667940

### Steps To Reproduce

https://app.intercom.com/a/apps/y10e7138/inbox/inbox/conversation/164629100140873#part_id=comment-164629100140873-15716667940

### Public Sample App

_No response_

### Version

1.6.6

### PR DESCRIPTION
**Pull request by rimildeyjsr**
## Context/Description:

A user pointed out that on copying the JS object from the JS editor page, we are appending "copy" to the name, but on copying it from the explorer page, we don't do this.

We are removing the "copy" word to remove this discrepancy in the behaviour of the same action from two different menus!

Same change is being made for actions as well!

Screen recording of fix - https://user-images.githubusercontent.com/10229595/178429380-7cc31fce-77f1-4b48-8657-e59eb95537f6.mov

Original report - https://app.intercom.com/a/apps/y10e7138/inbox/inbox/conversation/164629100140873#part_id=comment-164629100140873-15716667940

Fixes #15143 

## Type of change

- Bug fix (non-breaking change which fixes an issue)

## How Has This Been Tested?
**Test plan**
- [ ] https://github.com/appsmithorg/TestSmith/issues/1941

- Cypress tests

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Review by Rishabh-Rathod — DISMISSED** [review]
LGTM. 

Just to track this effort in current sprint can you please also 
- Create an issue for it and tag as `fixes #xxxxx` in the PR.
- Add sprint + estimate + assignee to that issue this way our velocity tracking will be more accurate

**Comment by unknown1337** [comment]
Thanks for creating and solving :)

**Review by ohansFavour — COMMENTED** [review]
Can we also fix the discrepancy in the labels?

Let's stick to one ( `copy to page` or `copy to a page` )

**Review by Rishabh-Rathod — DISMISSED** [review]
LGTM

File: app/client/cypress/integration/Smoke_TestSuite/ClientSideTests/JsObjects/JsObject_Copy_to_different_page_spec.js [hunk 1]
```diff
@@ -0,0 +1,41 @@
+describe("Copy JS objects to different pages", () => {
+  it("copies JS object to a different page from the additional menu in Queries/JS section", () => {
+    cy.Createpage("Page2");
+
+    cy.get(".t--entity-item:contains(Page1)")
```
[the remark this answers is not in the document]
Comment (rimildeyjsr) [reply app/client/cypress/integration/Smoke_TestSuite/ClientSideTests/JsObjects/JsObject_Copy_to_different_page_spec.js]: done

**Review by Rishabh-Rathod — APPROVED** [review]
LGTM

**Comment by ramsaptami** [comment]
Tested ok on deploy preview

**Review by ohansFavour — DISMISSED** [review]
Looks good

**Review by Rishabh-Rathod — APPROVED** [review]
LGTM

File: app/client/cypress/integration/Smoke_TestSuite/ClientSideTests/ActionObjects/ActionObject_copy_to_different_page_spec.js [hunk 2]
```diff
@@ -0,0 +1,41 @@
+const commonLocators = require("../../../../locators/commonlocators.json");
+import { ObjectsRegistry } from "../../../../support/Objects/Registry";
+let entityExplorer = ObjectsRegistry.EntityExplorer;
+
+describe("Copy Action objects to different pages", () => {
```
Comment (Aishwarya-U-R) [inline app/client/cypress/integration/Smoke_TestSuite/ClientSideTests/ActionObjects/ActionObject_copy_to_different_page_spec.js:5]: 1. Query copy missing
2. There can be one spec file with Bug id as spec name & all of 6 cases into same file
3. Spec can be places under BugTests
4. Index the 'it' functions
5. If possible write tests in .ts format

**Review by Aishwarya-U-R — COMMENTED** [review]
@rimildeyjsr - few comments below:
1. Do not use extra waits - use only if need be
2. Previously test script was only copying - but the bug is also about copying & making sure no 'Copy' keyword is added - have added the validation now, but take care of such core validations from next script onwards.

File: app/client/cypress/locators/JSEditor.json [hunk 3]
```diff
@@ -8,7 +8,8 @@
     "propertyList": ".t--entity-property",
     "delete": ".single-select >div:contains('Delete')",
     "deleteConfirm": ".single-select >div:contains('Are you sure?')",
-    "popover": "//*[local-name()='g' and @id='Icon/Outline/more-vertical']"
+    "popover": "//*[local-name()='g' and @id='Icon/Outline/more-vertical']",
+    "jsActionMenu": ".t--more-action-menu"
```
Comment (Aishwarya-U-R) [inline app/client/cypress/locators/JSEditor.json:12]: Remove the unused locators

File: app/client/cypress/locators/commonlocators.json [hunk 4]
```diff
@@ -182,5 +182,6 @@
   "saveThemeBtn": ".t--save-theme-btn",
   "selectThemeBackBtn": ".t--theme-select-back-btn",
   "themeAppBorderRadiusBtn": ".t--theme-appBorderRadius",
-  "codeEditorWrapper": ".unfocused-code-editor"
-}
\ No newline at end of file
+  "codeEditorWrapper": ".unfocused-code-editor",
+  "expandMore": "[name='expand-more']"
```
Comment (Aishwarya-U-R) [inline app/client/cypress/locators/commonlocators.json:186]: Remove the unused locators

File: app/client/cypress/locators/JSEditor.json (hunk 3 above)
Comment (rimildeyjsr) [reply app/client/cypress/locators/JSEditor.json:12]: done

File: app/client/cypress/locators/commonlocators.json (hunk 4 above)
Comment (rimildeyjsr) [reply app/client/cypress/locators/commonlocators.json:186]: done

**Review by Aishwarya-U-R — COMMENTED** [review]
Cypress looks good!

**Review by Aishwarya-U-R — COMMENTED** [review]
Please merge with latest release & start new run on this!

**Review by Aishwarya-U-R — DISMISSED** [review]

**Comment by AnandiKulkarni** [comment]
Issue - 
Name is getting updated for already copied query/api/jsboject differently while dobject/query oing it from JSeditor and explore 

1. Create a JSobject/query/api on page 1 and copy it to page 2 from explorer.
2. Copy again same JSobj/api/query from explorer - It is created as - #Name#+Copy
Perform above step through JSeditor  - Second Copy is created as - #Name#+1

image

**Comment by Rishabh-Rathod** [comment]
Please re-request review once the PR is ready to review with all tests passing. 
( _Just to avoid false review requested GitHub notification, I have removed the current request._ )

**Comment by AnandiKulkarni** [comment]
@rimildeyjsr two jsobject with same name is getting created 

Uploading Screen Recording 2022-09-20 at 10.05.02 AM.mov…

**Comment by rimildeyjsr** [comment]
We will be looking into this issue more closely and have a unified solution for the copy/move name conflicts and naming conventions. Closing this for now!

**Pull request closed without merging**